test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -2587,6 +2587,40 @@ BOOST_AUTO_TEST_CASE(function_modifier_overriding)
 	BOOST_CHECK(callContractFunction("f()") == encodeArgs(false));
 }
 
+BOOST_AUTO_TEST_CASE(shadow_function_by_accessor)
+{
+	char const* sourceCode = R"(
+		contract A {
+			function f() returns (uint) { return 1; }
+		}
+		contract B is A {
+			uint256 public f;
+			function B() {
+				f = 2;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "B");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(2));
+}
+
+BOOST_AUTO_TEST_CASE(implement_interface_by_accessor)
+{
+	char const* sourceCode = R"(
+		contract A {
+			function f() returns (uint);
+		}
+		contract B is A {
+			uint256 public f;
+			function B() {
+				f = 2;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "B");
```

Comment:
So this test need to be copied.   In one variant `A.f` should be `external` (which should succeed).  The original should not compile.